File: src/common/py/core/messaging/event.py
import typing
from dataclasses import dataclass

from .message import Message


@dataclass(frozen=True, kw_only=True)
class Event(Message):
    """
    An event message.

    Events are simple notifications that do not require a reply nor will *execute* anything.
    """
    pass


EventType = typing.TypeVar("EventType", bound=Event)

File: src/common/web/core/messaging/Event.ts
import { Message, type MessageCategory } from "./Message";

/**
 * An event message.
 *
 * Events are simple notifications that do not require a reply nor will *execute* anything.
 */
export class Event extends Message {
    public static readonly Category: MessageCategory = "Event";

    /**
     * Gets the global message category.
     */
    public get messageCategory(): MessageCategory {
        return Event.Category;
    }
}
